### PR TITLE
Feature/makerdao lending markets support

### PR DESCRIPTION
**Pull request by gitcoindev**
Hi @TheMoneyLisa it seems I have finally something that works :smile: I learnt a bit Kotlin on the way.

I would be grateful if you had a look and check if this is going in the right direction and what is missing to merge.

I tested `./run-protocol.sh makerdao` with `lending/all-markets` on 8081, I am pasting the partial output below:

```jsonc
{
"id":"makerdao-ethereum-0x03ae53b33feeac1222c3f372f32d37ba95f0f099",
"name":"UNIV2USDCETH-A",
"protocol":{"name":"MAKERDAO","logo":"https://github.com/defitrack/data/raw/master/logo/protocol/makerdao.png","slug":"makerdao","primitives":["LENDING"],"website":"https://makerdao.com/"},
"network":{"name":"ETHEREUM","logo":"https://github.com/defitrack/data/raw/master/logo/network/ethereum.png","chainId":1},
"token":{"address":"0x03ae53b33feeac1222c3f372f32d37ba95f0f099","name":"unknown","decimals":18,"symbol":"unknown","logo":"https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/0x03Ae53B33FeeAc1222C3f372f32D37Ba95f0F099/logo.png","type":"SINGLE"},
"rate":1.5,
"poolType":"makerdao-lending",
"marketSize":null,
"prepareInvestmentSupported":false},
{"id":"makerdao-ethereum-0x08638ef1a205be6762a8b935f5da9b700cf7322c",
"name":"ETH-B",
"protocol":{"name":"MAKERDAO","logo":"https://github.com/defitrack/data/raw/master/logo/protocol/makerdao.png","slug":"makerdao","primitives":["LENDING"],"website":"https://makerdao.com/"},
"network":{"name":"ETHEREUM","logo":"https://github.com/defitrack/data/raw/master/logo/network/ethereum.png","chainId":1},
"token":{"address":"0x08638ef1a205be6762a8b935f5da9b700cf7322c","name":"unknown","decimals":18,"symbol":"unknown","logo":"https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/0x08638eF1A205bE6762A8b935F5da9b700Cf7322c/logo.png","type":"SINGLE"},
"rate":4.0,
"poolType":"makerdao-lending",
"marketSize":null,
"prepareInvestmentSupported":false}
...
```

Also, feel free to directly edit the pr.

### CODE REVIEW

**Comment by TheMoneyLisa** [comment]
Awesome work, I'll take some time to review the code, but I'm amazing by the speed you delivered this. 

Are there any things in particular you immediately think of that I should add in the documentation/code? Seeing you're one of the first contributors, your insights will be valuable. 

You'll hear from me soon (here and on gitcoin ofc ;) )

**Comment by gitcoindev** [comment]
Thank you! About the documentation and code, for future contributors it would be useful to extend the https://docs.defitrack.io/developers/building-and-running/defitrack-core section on how to test the service on a few examples. It should be easy as you'd explain to a 5 year old :)  Also, I tried to use Android Studio for Kotlin, but the screenshots on how to setup Intellij somehow do not match, so I ended up using VS code, with a few issues though as I could not easily enable jumping from declaration to implementation. Therefore, instructions on how to set up Defitrack's recommended IDE, be it Intellij or VSCode to enable all the needed features for development and e.g. recommended plugins for code formatting would be a huge help.

File: defitrack-rest/defitrack-protocol-services/defitrack-makerdao/src/main/java/io/defitrack/protocol/makerdao/lending/market/MakerDAOEthereumLendingMarketProvider.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                try {
+                    val token = erc20Resource.getTokenInformation(getNetwork(), it.id)
+
+                    LendingMarket(
+                        id = "makerdao-ethereum-${it.id}",
+                        network = getNetwork(),
+                        protocol = getProtocol(),
+                        address = it.id,
+                        name = it.name,
+                        rate = it.rates[0].rate.toBigDecimal(),
```
Comment (TheMoneyLisa) [inline defitrack-rest/defitrack-protocol-services/defitrack-makerdao/src/main/java/io/defitrack/protocol/makerdao/lending/market/MakerDAOEthereumLendingMarketProvider.kt:37]: make sure this doesn't crash if rates was empty. `.firstOrNull()` could be used here.

**Review by TheMoneyLisa — CHANGES_REQUESTED** [review]

**Comment by TheMoneyLisa** [comment]
Added 1 tiny change request. Merging this after this is fixed. 

Thanks for the feedback. I'll be adding that to the documentation! Already improved the docs a bit using your PR.

File: defitrack-rest/defitrack-protocol-services/defitrack-makerdao/src/main/java/io/defitrack/protocol/makerdao/lending/market/MakerDAOEthereumLendingMarketProvider.kt (hunk 1 above)
Comment (gitcoindev) [reply defitrack-rest/defitrack-protocol-services/defitrack-makerdao/src/main/java/io/defitrack/protocol/makerdao/lending/market/MakerDAOEthereumLendingMarketProvider.kt]: done.

**Comment by TheMoneyLisa** [comment]
Awesome! I don't think you've "started work" on gitcoin yet. Not sure if everything will go as planned If I just merge? We'll sort it out if it doesn't :D

**Pull request merged**